openedx/core/djangoapps/catalog/utils.py

```diff
@@ -101,13 +137,8 @@ def get_program_types(name=None):
         list of dict, representing program types.
         dict, if a specific program type is requested.
     """
-    catalog_integration = CatalogIntegration.current()
-    if catalog_integration.enabled:
-        try:
-            user = catalog_integration.get_service_user()
-        except ObjectDoesNotExist:
-            return []
-
+    user, catalog_integration = check_catalog_integration_and_get_user(error_message_field='Program types')
```

Comment:
seems a little strange not to use the exact field name here ('program_types') but I see that this keeps things consistent with the other cases such as 'Course runs', and we don't want to change those error messages from what they are currently outputting, so this looks good